modules/prohibited-crd-upgrades.adoc

```diff
@@ -0,0 +1,39 @@
+// Module included in the following assemblies:
+//
+// * extensions/ce/crd-upgrade-safety.adoc
+
+:_mod-docs-content-type: REFERENCE
+
+[id="prohibited-crd-upgrades_{context}"]
+= Prohibited CRD upgrade changes
+
+The following changes to an existing custom resource definition (CRD) are caught by the CRD upgrade safety preflight check and prevent the upgrade:
+
+- A new required field is added to an existing version of the CRD
+- An existing field is removed from an existing version of the CRD
+- An existing field type is changed in an existing version of the CRD
+- A new default value is added to a field that did not previously have a default value
+- The default value of a field is changed
+- An existing default value of a field is removed
+- New enum restrictions are added to an existing field which did not previously have enum restrictions
+- Existing enum values from an existing field are removed
+- The minimum value of an existing field is increased in an existing version
+- The maximum value of an existing field is decreased in an existing version
```

Comment:
Would it help to highlight the constraints in which this applies? This applies for things like `minimum`, `minItems`, `minLength`, and `minProperties` field constraints. 

Same goes for maximum (just replace `min` with `max` in all the above mentioned constraints)